source/common/upstream/outlier_detection_impl.cc

```diff
@@ -249,9 +249,18 @@ DetectorConfig::DetectorConfig(const envoy::config::cluster::v3::OutlierDetectio
                                           DEFAULT_ENFORCING_CONSECUTIVE_LOCAL_ORIGIN_FAILURE))),
       enforcing_local_origin_success_rate_(static_cast<uint64_t>(
           PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, enforcing_local_origin_success_rate,
-                                          DEFAULT_ENFORCING_LOCAL_ORIGIN_SUCCESS_RATE))),
-      max_ejection_time_ms_(static_cast<uint64_t>(
-          PROTOBUF_GET_MS_OR_DEFAULT(config, max_ejection_time, DEFAULT_MAX_EJECTION_TIME_MS))) {}
+                                          DEFAULT_ENFORCING_LOCAL_ORIGIN_SUCCESS_RATE))) {
+  // Check if max_ejection_time is specified.
```

Comment:
cc @envoyproxy/api-shepherds 

Please update API comments since default behavior after this change is more complex than currently documented.  There were some comments in the original PR about the change being technically breaking behavior; was the bad interaction with configured base ejection time the concern or something else?  https://github.com/envoyproxy/envoy/pull/14235/files#r539583484

Reply:
Updated the description of the parameter in proto file. Normally outlier detector checks if max_ejection is larger than base_ejection time. If max_ is smaller, the config is rejected. The corner case was when max_ejection was not specified (legacy config was used from before the max_ejection was introduced) and base_ejection time was larger than max_ejection default. The config was rejected and broke upgrade path in istio.